src/ember-install-addon.js

```diff
@@ -1,88 +0,0 @@
-'use strict';
-
-const execa = require('execa');
-const semver = require('semver');
-const debug = require('./debug');
-
-// not fixed yet
-// https://github.com/ember-cli/ember-cli/issues/8937
-const buggyEmberCliRange = '*';
-
-function ember(args, {
-  cwd,
-  stdin = 'inherit'
-}) {
-  debug(`ember ${args.join(' ')}`);
-
-  let ps = execa('ember', args, {
-    cwd,
-    preferLocal: true,
-    stdio: [stdin, 'pipe', 'inherit']
-  });
-
-  ps.stdout.pipe(process.stdout);
-
-  return ps;
-}
-
-async function getEmberCliVersion({
-  cwd
-}) {
-  // Perhaps `npm ls` or even `require` would be faster/better...
-  let { stdout } = await ember(['v'], { cwd });
-
-  return stdout.match(/^ember-cli: (.+)$/m)[1];
-}
-
-async function emberInstallAddon({
-  cwd,
-  addonNameOverride,
-  packageName,
-  version,
-  blueprintPath,
-  stdin
-}) {
-  let addon;
-
-  if (addonNameOverride) {
-    addon = addonNameOverride;
-    if (version && !blueprintPath) {
-      addon += `@${version}`;
-    }
-  }
-
-  if (!addon) {
-    addon = blueprintPath;
-  }
-
-  if (!addon) {
-    addon = `${packageName}@${version}`;
-  }
-
-  let install = ember(['i', addon], { cwd, stdin });
-
-  let generate;
-
-  if (!blueprintPath) {
```

Comment:
I don't see this logic block in the new version. Is that going to be a problem?

Reply:
Let me know if I missed a use case.

I moved the responsibility of choosing which blueprint to use up to the parent function that calls  `installAndGenerateBlueprint`.

`install` command will default to the addon name if a `blueprint` param is not passed by the user.

line 45-53 of src/install.js
```
  let { ps } = await installAndGenerateBlueprint({
    cwd,
    addonNameOverride: addon,
    packageName,
    blueprintPath: path
    blueprintName: _blueprintName || addon,
    blueprintPath: path,
    packageManager: isYarnProject ? 'yarn' : 'npm'
  });
```



`installAddonBlueprint` also will pass a `blueprintName` [param](https://github.com/ember-cli/ember-cli-update/pull/1084/files/e64dcf32b896b312851f49b29e810a4775e7062a)

line 275-284 of src/get-start-and-end-command.js
```
let { ps } = await installAndGenerateBlueprint({
    cwd: projectRoot,
    packageName: blueprint.packageName,
    version: blueprint.version,
    blueprintPath: blueprint.path,
    stdin: 'pipe'
    blueprintName: blueprint.name,
    blueprintOptions: blueprint.options || [],
    stdin: 'pipe',
    packageManager
  });
```